lib/content-services/src/lib/search/components/search-chip-autocomplete-input/search-chip-autocomplete-input.component.ts

```diff
@@ -88,33 +119,31 @@ export class SearchChipAutocompleteInputComponent implements OnInit, OnDestroy {
     }
 
     selected(event: MatAutocompleteSelectedEvent) {
-        if (!this.isAdded(event.option.viewValue)) {
-            this.selectedOptions.push(event.option.viewValue);
-            this.optionInput.nativeElement.value = '';
-            this.formCtrl.setValue(null);
-            this.optionsChanged.emit(this.selectedOptions);
-        }
+        this.selectedOptions.push(event.option.value);
+        this.optionInput.nativeElement.value = '';
+        this.formCtrl.setValue('');
+        this.optionsChanged.emit(this.selectedOptions);
     }
 
-    private filter(value: string): string[] {
+    private filter(value: string): AutocompleteOption[] {
         const filterValue = value.toLowerCase();
-        return this.autocompleteOptions.filter(option => option.toLowerCase().includes(filterValue)).slice(0, 15);
+        return this.autocompleteOptions.filter(option => option.value.toLowerCase().includes(filterValue)).slice(0, 15);
     }
 
-     private isAdded(value: string): boolean {
-        return this.selectedOptions.includes(value);
+    private isAdded(value: string): boolean {
+        return this.selectedOptions.some(option => option.value.toLowerCase() === value.toLowerCase());
```

Comment:
can we call toLowerCase for value before some function? No need to make it lower case for each iteration